Bundle/DependencyInjection/Configuration.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\PsrHttpMessage\Bundle\DependencyInjection;
+
+use Laminas\Diactoros\RequestFactory;
+use Laminas\Diactoros\ServerRequestFactory;
+use Nyholm\Psr7\Factory\Psr17Factory;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+
+/**
+ * @author Alexander M. Turek <me@derrabus.de>
+ */
+final class Configuration implements ConfigurationInterface
+{
+    public function getConfigTreeBuilder(): TreeBuilder
+    {
+        $treeBuilder = new TreeBuilder('psr_http_message');
+        $rootNode = $treeBuilder->getRootNode();
+
+        $rootNode->children()
+            ->arrayNode('message_factories')
+                ->{class_exists(Psr17Factory::class) || class_exists(ServerRequestFactory::class) ? 'canBeDisabled' : 'canBeEnabled'}()
+                ->children()
+                    ->enumNode('implementation')
+                        ->info('The PSR-7 implementation to use. By default, the bundle will configure the Nyholm implementation if it\'s available.')
+                        ->values(['nyholm', 'diactoros'])
+                        ->defaultValue(class_exists(Psr17Factory::class) || !class_exists(ServerRequestFactory::class) ? 'nyholm' : 'diactoros')
```

Comment:
this makes nyholm/psr7 a mandatory dependency when the bundle is used, isn't it?
the default might be null instead, meaning it's fine if no service are wired

Reply:
Only when this branch is enabled I think, since a few lines above we have:

    ->{class_exists(Psr17Factory::class) || class_exists(ServerRequestFactory::class) ? 'canBeDisabled' : 'canBeEnabled'}()

Reply:
@Jean85 That was the idea, yes. Maybe I should add a CI job that tests how the bundle behaves after removing the PSR-7 implementations?

Reply:
Yes, definitely.

Reply:
Done.